src/test/scala/com/fortysevendeg/android/sqlite/resultset/SQLDroidResultSetDataSpec.scala

```diff
@@ -0,0 +1,604 @@
+        sqlDroid.getShort(columnInteger) must throwA[SQLException]
+      }
+
+  }
+
+  "getObject" should {
+
+    "return the String when fetching the String value" in
+      new WithMatrixCursor with WithData {
+        cursor.moveToNext()
+        val o = sqlDroid.getObject(columnString)
+        o must haveClass[java.lang.String]
+        o shouldEqual rows.head(columnString - 1)
+      }
+
+    "return the Integer when fetching the Integer value" in
+      new WithMatrixCursor with WithData {
+        cursor.moveToNext()
+        val o = sqlDroid.getObject(columnInteger)
+        o must haveClass[java.lang.Integer]
+        o shouldEqual rows.head(columnInteger - 1)
+      }
+
+    "return the Float when fetching the Float value" in
+      new WithMatrixCursor with WithData {
+        cursor.moveToNext()
+        val o = sqlDroid.getObject(columnFloat)
+        o must haveClass[java.lang.Float]
+        o shouldEqual rows.head(columnFloat - 1)
+      }
+
+    "return the Blob when fetching the byte array value" in
+      new WithMatrixCursor with WithData {
+        cursor.moveToNext()
+        val o = sqlDroid.getObject(columnBytes)
+        o must haveClass[SQLDroidBlob]
+      }
+
+    "return null when the field is null" in
+      new WithMatrixCursor with WithData {
+        cursor.moveToNext()
+        sqlDroid.getObject(columnNull) must beNull
+      }
+
+  }
+
+  "getNCharacterStream" should {
+
+    "throws a SQLFeatureNotSupportedException" in new WithCursorMocked {
+      sqlDroid.getNCharacterStream(1) must throwA[SQLFeatureNotSupportedException]
+    }
+
+  }
+
+  "getRef" should {
+
+    "throws a SQLFeatureNotSupportedException" in new WithCursorMocked {
+      sqlDroid.getRef(1) must throwA[SQLFeatureNotSupportedException]
+    }
+
+  }
+
+  "getBlob" should {
+
+    "return a right Blob when the data is stored in a byte array" in
+      new WithMatrixCursor with WithData {
+        cursor.moveToNext()
+        val value = new String(rows.head(columnBytes - 1).asInstanceOf[SArray[Byte]])
+        inputStreamToString(sqlDroid.getBlob(columnBytes).getBinaryStream) shouldEqual value
+      }
+
+    "return null when the field is null" in
+      new WithMatrixCursor with WithData {
+        cursor.moveToNext()
+        sqlDroid.getBlob(columnNull) must beNull
+      }
+
+    "throws SQLException when the cursor is closed" in
+      new WithMatrixCursor {
+        cursor.close()
+        sqlDroid.getBlob(columnBytes) must throwA[SQLException]
+      }
+
+  }
+
+  "getDate" should {
+
+    "return a right Date when is stored in an number" in
+      new WithMatrixCursor with WithData {
+        val date = new java.sql.Date(rows.head(columnInteger - 1).asInstanceOf[Int])
+        cursor.moveToNext()
+        sqlDroid.getDate(columnInteger) shouldEqual date
+      }
+
+    "return a right Date when is stored in a string with valid format" in
+      new WithMatrixCursor {
+        val date = new java.sql.Date(Random.nextInt(100))
```

Comment:
AFAIK `java.sql.Date` receives a _long_ instead of a _int_ in the constructor, right?